scripts/proto/lbann_proto.py

```diff
@@ -41,85 +42,78 @@ def _add_to_module_namespace(stuff):
     for k, v in stuff.items():
         g[k] = v
 
-class Module:
-    """Basic building block for layers (or sets of layers).
+class Layer:
+    """Base class for layers."""
 
-    Inherit from this for constructing neural networks. Modules are constructed
-    by calling an instance (e.g. module_instance(x)) to make x a parent of
-    module_instance.
+    num_layers = 0  # Static counter, used for default layer names
 
-    Internally, that will invoke the add_parent method, which can be overriden
-    by child modules to customize adding parents.
-
-    """
-
-    def __init__(self, name, data_layout):
-        self.name = name
-        self.data_layout = data_layout
-        self.parents = []
-        self.children = []
+    def __init__(self, name='', inputs=[], outputs=[],
+                 weights=[], data_layout=''):
+        Layer.num_layers += 1
+        self.name = name if name else 'layer{0}'.format(Layer.num_layers-1)
+        self.inputs = []
+        self.outputs = []
         self.weights = []
+        self.data_layout = data_layout
 
-    def __call__(self, parent):
```

Comment:
I think we should keep the callable syntax for chaining layers together.

Reply:
Done.